lib/Parse/ParseStmt.cpp

```diff
@@ -1417,6 +1417,13 @@ ParserResult<Stmt> Parser::parseStmtIf(LabeledStmtInfo LabelInfo) {
         return recoverWithCond(Status, Condition);
     }
 
+    if (Tok.is(tok::kw_else)) {
+      SourceLoc ElseLoc = Tok.getLoc();
+      diagnose(ElseLoc, diag::unexpected_else_after_if)
+        .fixItRemove(SourceRange(ElseLoc));
```

Comment:
Nitpick: conventionally we don't bother making the conversion from SourceLoc to SourceRange explicit for fix-its. "Remove the token at this location" makes just as much sense.